shop3/looping-tasks/loop-extensions.lisp

```diff
@@ -258,6 +296,85 @@
   ;; Get the inner construct values
   (second (assoc key task-body)))
 
+
+(defmethod expand-conditional-body ((body-key (eql :if))
+				    task-body domain state
+				    in-unifier search-state)
+  (declare (ignorable search-state))
+  ;; TASK-BODY is the form (:IF (:COND ....) (:ORDERED ...) (:ELSE...))
+  (let* ((if-condition (block-body-condition :cond task-body))
+         (unifiers
+	   (find-satisfiers if-condition state
+			    :domain domain))
+         reductions)
+    (if unifiers
+	(iter
+	 (for u in unifiers)
+	 (let* ((u1 (compose-substitutions in-unifier u))
+		(then-body
+		  (apply-substitution
+		   (block-body-item :ordered task-body)
+		   u1)))
```

Comment:
If we are going to track dependencies, we need to do something more like:

`(multiple-value-bind (unifiers dependencies) (find-satisfiers...))`

I see that I need to fix the docstring for `find-satisfiers`, btw.

Then we may need to do `answer-set-union` instead of just `compose-substitutions` to merge dependencies in, as well.  *Possibly* this should be returning new dependencies as well as reductions?